File: libs/php/params.php
<?php

/**
 * @file params.php
 * @brief IPP project 1
 * @author Marek Gergel (xgerge01)
 * @date 2023-03-11
 * @details
 * Parameters parser class
 */

require_once 'error.php';

class Params
{
   private $help = "This script parses IPPcode23 code from STDIN (or source file) and outputs XML representation to STDOUT\n" . "Usage: php parse.php [--source=file]\n" . "--source=file - source file to parse\n" . "--help - print this help\n";
   public $source = STDIN;

   public function __construct($argv)
   {
      //parse script parameters
      for ($i = 1; $i < count($argv); $i++) {
         if (preg_match("/^--help$/", $argv[$i])) {

            if (count($argv) > 2) error_exit(ErrorCodes::ERR_PARAMS, "Parameter $argv[$i] must be used alone\n");
            echo $this->help;
            exit(0);
         } else if (preg_match("/^--source=\"?.+\"?$/", $argv[$i])) {

            $this->source = preg_replace("/^--source=\"?(.+)\"?$/", "$1", $argv[$i]);
         } else {

            error_exit(ErrorCodes::ERR_PARAMS, "Unknown parameter $argv[$i]\n");
         }
      }
   }
}

?>

File: libs/python/params.py
# IPP 2023, project 2
# Author: Marek Gergel, xgerge01
# File: params.py
# Description: Script parameter parsing
# Date: 2023-04-07

import argparse
import sys
from . import error 

source=""
input=""

def parse():
    parser = argparse.ArgumentParser(description='Interpret script for IPPcode23 in XML format', add_help=False,)
    parser.add_argument('--help', action='store_true', help='show this help message and exit')
    parser.add_argument('--source', metavar='filename', type=str, help='the source file with the source code IPPcode23 in XML format')
    parser.add_argument('--input', metavar='filename', type=str, help='the input file with input data')

    args = parser.parse_args()
    if args.help:
        if len(sys.argv) == 2:
            parser.print_help()
            exit(0)
        else:
            error.exit(error.code.ERR_PARAMS, 'Parameter \'--help\' must be used alone\n')

    if not args.source and not args.input:
        error.exit(error.code.ERR_PARAMS, 'Parameter \'--source\' or \'--input\' have to be specified\n')

    global source
    if args.source:
        source = args.source
    else:
        source = sys.stdin

    global input
    if args.input:
        input = args.input
    else:
        input = sys.stdin
